packages/components/src/tree-grid/stories/index.story.tsx

```diff
@@ -16,7 +16,8 @@ import { Button } from '../../button';
 import InputControl from '../../input-control';
 
 const meta: Meta< typeof TreeGrid > = {
-	title: 'Components (Experimental)/TreeGrid',
+	title: 'Components (Experimental)/Navigation/TreeGrid',
```

Comment:
Another thing (cc @ciampo @mirka): experimental components will fall in their own section down below, meaning they will have their own Navigation directory. Is that desired? Should we consider marking the experimental components just with badges, like `CustomSelectControlV2` is marked WIP?

Reply:
> experimental components will fall in their own section down below, meaning they will have their own Navigation directory.

Waiting on this until confirmation of above.

Cc: @tyxla

Reply:
I've been discussing this with @ciampo, and we concluded that it's a problem that can be solved separately. 

In the future we may remove the "Experimental" distinction from the Storybook component list, and those categories that we're introducing will be on the top level. That being said, we could leave things as they are and ship this PR, and I've opened #66805 to discuss this separately.